internal/services/postgres/postgresql_server_resource.go

```diff
@@ -446,6 +446,15 @@ func resourcePostgreSQLServer() *pluginsdk.Resource {
 				}
 				return false
 			}),
+			pluginsdk.ForceNewIfChange("create_mode", func(ctx context.Context, old, new, meta interface{}) bool {
+				oldMode := old.(postgresql.CreateMode)
+				newMode := new.(postgresql.CreateMode)
```

Comment:
these'll need to be:

```suggestion
				oldMode := postgresql.CreateMode(old.(string))
				newMode := postgresql.CreateMode(new.(string))
```